### ISSUE TITLE
PDF Mixed Page Orientation Incorrect

### ISSUE THREAD

**Issue by d-j-m-0** [trevorsandy/lpub3d#59]
### PDF generated by LPub3D has incorrect page orientations
When a mpd file contains multiple page orientations (i.e. landscape and portrait), although LPub3D displays them (mostly) with the correct orientation, the resulting generated PDF file has incorrect page orientations.

### Your environment
* Version of LPub3D - LPub3D 2.2.2.0.848 (12 04 2018 09:04:29)
* Operating system - Windows 10, 64 bit, Processor Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz, 2496 Mhz, 2 Core(s), 4 Logical Processor(s)

### Configuration
* LPub3D installation path
  - Windows: `D:\Program Files\LPub3D'
* Application data path
  - Windows: `%USERPROFILE%\AppData\Local\LPUb3D Software\LPub3D\`

### Logs
N/A

### Steps to reproduce

1. Rename the attached file from landscape_portrait_mpd.txt to landscape_portrait.mpd
[landscape_portrait_mpd.txt](https://github.com/trevorsandy/lpub3d/files/2011137/landscape_portrait_mpd.txt)
2. Open the attached mpd file landscape_portrait.mpd using LPub3D
3. Page through the file. The third page is shown with a portrait page orientation. All other pages are shown with a landscape page orientation. This is as expected.
4. Export the instructions to a PDF file
5. Open the PDF file. All pages have a portrait orientation (as per attached PDF).
[landscape_portrait.pdf](https://github.com/trevorsandy/lpub3d/files/2011138/landscape_portrait.pdf)

Additional comment:
The parts list on the page with the portrait orientation is placed off to the left of the page within LPub3d. Refer to the attached screenshot.
![landscape_portrait](https://user-images.githubusercontent.com/37604323/40148831-455eed8e-59c4-11e8-8917-0588fc5e1793.png)

### Expected behaviour
In the PDF, only the third page should have a portrait orientation. All other pages should have a landscape orientation.
In LPub3D, the parts list on the portrait page should appear on the page.

### Actual behaviour
All pages in the PDF have a portrait orientation.

### Workaround
Unknown.

### Solution suggestion
Unknown.

**Comment by trevorsandy** [comment]
Yep - some strange stuff going on here. Working on it now...

**Comment by trevorsandy** [comment]
Corrected.

**Comment by trevorsandy** [comment]
I've addressed both bugs described in this issue.

1. Mixed page orientation export does not output a mixed orientation file.
2. Mixed page orientation display positions the PLI outside the page bounds - see 'Additional comments' above.

Additionally, I've refactored the the page size routines with improved robustness and flexibility. Notably, it is no longer necessary to define both page width and height dimensions (e.g. 8.2677 x 11.6929) and the page size ID (e.g. A4), as in the first example below. Defining either will suffice, as in the second and third examples below. Any of the below page size definitions are now valid:

```
0 !LPUB PAGE SIZE GLOBAL 8.2677 11.6929 A4
0 !LPUB PAGE SIZE GLOBAL 8.2677 11.6929
0 !LPUB PAGE SIZE GLOBAL A4
```

As before, it is not necessary to include a PAGE SIZE meta when changing the page orientation. Only the page orientation meta e.g `0 !LPUB PAGE ORIENTATION GLOBAL LANDSCAPE` is necessary.

Also page width and height dimensions should always be defined in the order width then height. DO NOT flip the width and height dimensions to change the page orientation.
